admin-data-explorer-play-server/src/main/scala/net/wiringbits/webapp/utils/admin/repositories/daos/DatabaseTablesDAO.scala

```diff
@@ -151,11 +151,17 @@ object DatabaseTablesDAO {
           field <- fields
           fieldName = field.name
           data = {
-            if (settings.photoColumn.contains(fieldName)) {
-              // TODO: we're doing this operation two times (get id string), find another way
-              val rowId = resultSet.getString(settings.primaryKeyField)
-              s"$baseUrl/admin/images/$tableName/$rowId"
-            } else resultSet.getString(fieldName)
+            val maybe = settings.columnTypeOverrides.find(_._1 == fieldName)
+            maybe
+              .map { case (columnName, customDataType) =>
+                customDataType match {
+                  case CustomDataType.BinaryImage =>
+                    val rowId = resultSet.getString(settings.primaryKeyField)
```

Comment:
shouldn't we get a byte array instead of a string?

Reply:
No, the way this approach works is:
Before when we wanted to show the image, the binary data was loaded as an string and it was shown to the admin, this wasn't helpful at all. Now, when we're querying for a data (bytea) column, we can override it's type, if we want to display a photo on react-admin using a `ImageField` we need the actual photo name/photo url. That's why we're getting the id of the row and returning an string representation from where the photo is (photo URL), this URL is handled using a different endpoints for images.